sdk/log/simple_test.go

```diff
@@ -120,6 +128,247 @@ func TestSimpleProcessorConcurrentSafe(*testing.T) {
 	wg.Wait()
 }
 
+type errorExporter struct {
+	err error
+}
+
+func (e *errorExporter) Export(_ context.Context, _ []log.Record) error {
+	return e.err
+}
+
+func (*errorExporter) Shutdown(context.Context) error {
+	return nil
+}
+
+func (*errorExporter) ForceFlush(context.Context) error {
+	return nil
+}
+
+type failingMeterProvider struct {
+	noop.MeterProvider
+}
+
+func (*failingMeterProvider) Meter(name string, opts ...metric.MeterOption) metric.Meter {
+	return &failingMeter{Meter: noop.NewMeterProvider().Meter(name, opts...)}
+}
+
+type failingMeter struct {
+	metric.Meter
+}
+
+func (*failingMeter) Int64Counter(_ string, _ ...metric.Int64CounterOption) (metric.Int64Counter, error) {
+	return nil, errors.New("failed to create counter")
+}
+
+func TestSimpleProcessorSelfObservability(t *testing.T) {
+	originalMP := otel.GetMeterProvider()
+	defer otel.SetMeterProvider(originalMP)
+
+	t.Run("self observability disabled", func(t *testing.T) {
+		t.Setenv(x.SelfObservability.Key(), "")
+
+		e := new(exporter)
+		s := log.NewSimpleProcessor(e)
+
+		r := new(log.Record)
+		r.SetSeverityText("test")
+		_ = s.OnEmit(context.Background(), r)
+
+		require.True(t, e.exportCalled)
+		assert.Equal(t, []log.Record{*r}, e.records)
+	})
+
+	t.Run("self observability enabled without error", func(t *testing.T) {
+		t.Setenv(x.SelfObservability.Key(), "true")
+
+		reader := metricSDK.NewManualReader()
+		mp := metricSDK.NewMeterProvider(metricSDK.WithReader(reader))
+		otel.SetMeterProvider(mp)
```

Comment:
Since it's a global `MeterProvider`, I recommend restoring the global value after the unit tests have completed execution.

Reply:
Thanks for the suggestion! I've added a cleanup helper that restores the global MeterProvider after each subtest execution.